phpseclib/File/X509.php

```diff
@@ -2820,7 +2821,8 @@ public function setEndDate($date)
                 $date = new DateTime($date);
             }
 
-            $this->endDate = $date->format('D, d M Y H:i:s O', new DateTimeZone(@date_default_timezone_get()));
+            $date->setTimezone(new DateTimeZone(@date_default_timezone_get()));
+            $this->endDate = $date->format('D, d M Y H:i:s O');
```

Comment:
I think this is best added as a second parameter to the `DateTime` constructor. The `DateTime` constructor does accept two parameters, the second of which is `DateTimeZone`:

http://php.net/manual/en/datetime.construct.php

Also, without it being in the constructor, `$date = new DateTime('now');` returns a fatal error when the timezone hasn't been set in php.ini due to an exception not being caught.

It doesn't really serve the same point if set in `$date->setTimezone`.

Anyway, nice job on finding the issue! Altho I think I'll take a different route for fixing the issue, you found it - thanks!